operator/apis/lifecycle/v1alpha3/keptntaskdefinition_types.go

```diff
@@ -125,3 +132,50 @@ type KeptnTaskDefinitionList struct {
 func init() {
 	SchemeBuilder.Register(&KeptnTaskDefinition{}, &KeptnTaskDefinitionList{})
 }
+
+func (d KeptnTaskDefinition) SpecExists() bool {
+	return d.IsJSSpecDefined() || d.IsContainerSpecDefined()
+}
+
+func (d KeptnTaskDefinition) IsJSSpecDefined() bool {
+	return d.Spec.Function != nil
+}
+
+func (d KeptnTaskDefinition) IsContainerSpecDefined() bool {
+	return d.Spec.Container != nil
+}
+
+func (d KeptnTaskDefinition) IsVolumeMountPresent() bool {
+	return d.IsContainerSpecDefined() && d.Spec.Container.VolumeMounts != nil && len(d.Spec.Container.VolumeMounts) > 0
+}
+
+func (d KeptnTaskDefinition) GetVolumeSource() *v1.EmptyDirVolumeSource {
+	if d.IsContainerSpecDefined() {
+		quantity, ok := d.Spec.Container.Resources.Limits["memory"]
+		if ok {
+			return &v1.EmptyDirVolumeSource{
+				SizeLimit: &quantity,
+			}
+		}
+	}
+
+	return &v1.EmptyDirVolumeSource{
+		// Default 50% of the memory of the node, max 1Gi
+		SizeLimit: resource.NewQuantity(1, resource.Format("Gi")),
+		Medium:    v1.StorageMedium("Memory"),
+	}
+}
```

Comment:
in the refactoring, I would move this logic into the file that needs this business logic. I don't think it fits the CRD file